### DISCUSSION TITLE
Preloader is not smooth

### DISCUSSION THREAD

**Discussion by krishnagogada** [eclipse-theia/theia#9287] [Q&A]
I planned to keep Lottie animation as a loading view in a theia-ide, it is stuck with respective to API calls.

![112724652-662c8180-8f3a-11eb-8101-4e6d9a7c4129](https://user-images.githubusercontent.com/54623105/113124001-b3f30380-9232-11eb-96cd-2a3875e46a46.gif)

is there any relation between api's and Lottie animation frames?
How can I resolve this?

**Comment by vince-fugnitto** [comment]
@krishnagogada thank you for the discussion.

I'm not familiar with _Lottie_, but its likely that the busyness of the application starting leads to the animation not being smooth. 
If you can identify the main cause, it might be a well received change :+1: 

For additional information, I've noticed something similar when using an animated svg spinner:

https://user-images.githubusercontent.com/40359487/46867968-c2e06400-cdf4-11e8-834a-67cdb24253bf.gif

**Comment by marcdumais-work** [comment]
[Thomas recently mentioned](https://github.com/eclipse-theia/theia/issues/9514#issuecomment-993388406) at the `dev-meeting` about his analysis our performance issues, caused by encoding;/decoding of messages. At startup, when the preloader animation is displayed, is a busy time for the browser Theia application. A good chunk of the load is likely due to the issues Thomas mentions. 

TL;DR: (hopefully near) future enhancements to the encoding/decoding of Theia JSON-RPC communication is likely to positively impact preloader animation smoothness and many other things.
